### ISSUE TITLE
Feature Request: structured logs

### ISSUE THREAD

**Issue by gclawes** [Sonarr/Sonarr#6890]
### Is there an existing issue for this?

- [X] I have searched the existing open and closed issues

### Is your feature request related to a problem? Please describe

In modern logging pipelines, especially with container logging systems (docker/k8s) or even systemd journal structured logs, multi-line log events such as the example below often get recorded as many separate logging events in an indexing system.  For messages like stack traces, this makes it especially hard to parse a stack trace, as it gets split up into many events.

Example:
```
[Error] CommandExecutor: Error occurred while executing task ProcessMonitoredDownloads

[v3.0.10.1567] code = Busy (5), message = System.Data.SQLite.SQLiteException (0x800007AF): database is locked
database is locked
  at System.Data.SQLite.SQLite3.Step (System.Data.SQLite.SQLiteStatement stmt) [0x000ba] in <cf516e4846354910b3d60749c894b1bf>:0
  at System.Data.SQLite.SQLiteDataReader.NextResult () [0x00174] in <cf516e4846354910b3d60749c894b1bf>:0
  at System.Data.SQLite.SQLiteDataReader..ctor (System.Data.SQLite.SQLiteCommand cmd, System.Data.CommandBehavior behave) [0x0008e] in <cf516e4846354910b3d60749c894b1bf>:0
  at (wrapper remoting-invoke-with-check) System.Data.SQLite.SQLiteDataReader..ctor(System.Data.SQLite.SQLiteCommand,System.Data.CommandBehavior)
  at System.Data.SQLite.SQLiteCommand.ExecuteReader (System.Data.CommandBehavior behavior) [0x0000c] in <cf516e4846354910b3d60749c894b1bf>:0
  at System.Data.SQLite.SQLiteCommand.ExecuteNonQuery (System.Data.CommandBehavior behavior) [0x00006] in <cf516e4846354910b3d60749c894b1bf>:0
  at System.Data.SQLite.SQLiteCommand.ExecuteNonQuery () [0x00006] in <cf516e4846354910b3d60749c894b1bf>:0
  at Marr.Data.QGen.UpdateQueryBuilder`1[T].Execute () [0x0003b] in C:\BuildAgent\work\63739567f01dbcc2\src\Marr.Data\QGen\UpdateQueryBuilder.cs:157
  at NzbDrone.Core.Datastore.BasicRepository`1[TModel].SetFields (TModel model, System.Linq.Expressions.Expression`1[System.Func`2[TModel,System.Object]][] properties) [0x0002a] in C:\BuildAgent\work\63739567f01dbcc2\src\NzbDrone.Core\Datastore\BasicRepository.cs:246
  at NzbDrone.Core.Messaging.Commands.CommandRepository.End (NzbDrone.Core.Messaging.Commands.CommandModel command) [0x00000] in C:\BuildAgent\work\63739567f01dbcc2\src\NzbDrone.Core\Messaging\Commands\CommandRepository.cs:80
  at NzbDrone.Core.Messaging.Commands.CommandQueueManager.Update (NzbDrone.Core.Messaging.Commands.CommandModel command, NzbDrone.Core.Messaging.Commands.CommandStatus status, System.String message) [0x0005e] in C:\BuildAgent\work\63739567f01dbcc2\src\NzbDrone.Core\Messaging\Commands\CommandQueueManager.cs:247
  at NzbDrone.Core.Messaging.Commands.CommandQueueManager.Complete (NzbDrone.Core.Messaging.Commands.CommandModel command, System.String message) [0x00000] in C:\BuildAgent\work\63739567f01dbcc2\src\NzbDrone.Core\Messaging\Commands\CommandQueueManager.cs:185
  at NzbDrone.Core.Messaging.Commands.CommandExecutor.ExecuteCommand[TCommand] (TCommand command, NzbDrone.Core.Messaging.Commands.CommandModel commandModel) [0x000e0] in C:\BuildAgent\work\63739567f01dbcc2\src\NzbDrone.Core\Messaging\Commands\CommandExecutor.cs:96
  at (wrapper dynamic-method) System.Object.CallSite.Target(System.Runtime.CompilerServices.Closure,System.Runtime.CompilerServices.CallSite,NzbDrone.Core.Messaging.Commands.CommandExecutor,object,NzbDrone.Core.Messaging.Commands.CommandModel)
  at System.Dynamic.UpdateDelegates.UpdateAndExecuteVoid3[T0,T1,T2] (System.Runtime.CompilerServices.CallSite site, T0 arg0, T1 arg1, T2 arg2) [0x00035] in <69ada62907b24213a012734531df1db1>:0
  at (wrapper dynamic-method) System.Object.CallSite.Target(System.Runtime.CompilerServices.Closure,System.Runtime.CompilerServices.CallSite,NzbDrone.Core.Messaging.Commands.CommandExecutor,object,NzbDrone.Core.Messaging.Commands.CommandModel)
  at NzbDrone.Core.Messaging.Commands.CommandExecutor.ExecuteCommands () [0x00027] in C:\BuildAgent\work\63739567f01dbcc2\src\NzbDrone.Core\Messaging\Commands\CommandExecutor.cs:41
```
  
This gets recorded in a log aggregation system (Loki, in this example) as multiple events:
![Screenshot 2024-06-18 221623](https://github.com/Sonarr/Sonarr/assets/172517/384b3803-f9c6-41c7-b30a-160cc3014edd)

### Describe the solution you'd like

Structured logs (most prominently, [JSON](https://betterstack.com/community/guides/logging/json-logging/) and [logfmt](https://brandur.org/logfmt)) are often used in applications to encapsulate the log event in a machine-parseable data structure so complex fields (such as multi-line stack traces) can be parsed and extracted easily.  JSON  is especially good at this with properly escaped strings, though logfmt accomplishes this.

### Describe alternatives you've considered

It is possible to configure a log scraper like `promtail` to ingest multi-line messages using configs customized to the log message contents:
```
scrape_configs:
  - job_name: sonarr
    pipeline_stages:
      - docker: {}
      - multiline:
          firstline: '^\[[A-Za-z]+\]'
          max_wait_time: 3s
    kubernetes_sd_configs:
      - role: pod
    relabel_configs:
      - source_labels: [__meta_kubernetes_pod_label_name]
        target_label: job
```

This is however brittle and requires a lot of customization.  It would be very useful for the application to provide a standard parseable structure for log events.

**Comment by markus101** [comment]
Stack traces are always indented two lines. At this time no plans for customized or additional logging.
